File: api/models/usuario.py
"""Esquemas del usuario — ARCHIVO COMPARTIDO.

Dueño: Persona 1 (Login + Perfil).
"""

from pydantic import BaseModel, EmailStr


class Usuario(BaseModel):
    id: int
    nombre: str
    correo: EmailStr
    rol: str   # "administrador", "capturista", "consulta"
    area: str  # "Dirección General", "Procuración de Fondos"


class CredencialesEntrada(BaseModel):
    correo: EmailStr
    password: str


class TokenSalida(BaseModel):
    token: str
    tipo: str = "bearer"

File: ios/SistemaIngresos/Core/Model/Usuario.swift
//
//  Usuario.swift
//  Core / Model — ARCHIVO COMPARTIDO.
//  Lo consumen Login + Perfil y el resto de pantallas para permisos.
//

import Foundation

struct Usuario: Identifiable, Codable {
    let id: Int
    let nombre: String
    let correo: String
    let rol: String        // "ADMINISTRADOR", "TELEFONISTA", "RECOLECTOR"
    let area: String       // "Dirección General", "Procuración de Fondos"
}
